ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/XARecoveryModuleUnitTest.java

```diff
@@ -525,20 +533,17 @@ public XAResource[] getXAResources() throws Exception {
         // 1st pass: returns one xid (count is 1)
         xarm.periodicWorkFirstPass();
         // 2nd pass: throws an exception (count is 2)
-        xarm.periodicWorkSecondPass();
-        assertTrue(xarm.getContactedJndiNames().contains("test"));
```

Comment:
Which bit of code caused this assertion to fail?
Or have you removed it for some other reason?

Reply:
It fails because now the counts are not aligned to the logic implemented in the XAResourceWrapper. Let me fix that.

Reply:
the assertion failed because now the exception is thrown during the  periodicWorkFirstPass, where the list of Jndi is cleaned: see https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java#L187